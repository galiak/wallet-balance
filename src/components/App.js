import React from 'react';
import UserBalance from './UserBalance';
import BitcoinBalance from './BitcoinBalance';

const App = () => {
  return (
    <div className="ui segment">
      <h2>Wallet Balance</h2>
      <div className="ui divider"></div>
      <UserBalance />
      <div className="ui divider"></div>
      <BitcoinBalance />
      <div><a target="_blank" href="https://www.coindesk.com/price/bitcoin">Powered by CoinDesk</a></div>
    </div>
  );
}

export default App;