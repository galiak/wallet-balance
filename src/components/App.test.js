import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import '../setupTests';


describe('App', () => {
  const app = shallow(<App/>);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });

  it('contains a connected Wallet component', () => {
    // console.log(app.debug());

    expect(app.find('Connect(UserBalance)').exists()).toBe(true);
  });

  it('contains a connected BitcoinBalance component', () => {
    expect(app.find('Connect(BitcoinBalance)').exists()).toBe(true);
  })

  it('contains a link to the coindesk price page', () => {
    expect(app.find('a').props().href).toBe('https://www.coindesk.com/price/bitcoin');
  });
});