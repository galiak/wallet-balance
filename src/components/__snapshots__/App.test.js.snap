// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Wallet Balance
        </h2>,
        <div
          className="ui divider"
        />,
        <Connect(UserBalance) />,
        <div
          className="ui divider"
        />,
        <Connect(BitcoinBalance) />,
        <div>
          <a
            href="https://www.coindesk.com/price/bitcoin"
            target="_blank"
          >
            Powered by CoinDesk
          </a>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Wallet Balance",
        },
        "ref": null,
        "rendered": "Wallet Balance",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ui divider",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ui divider",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://www.coindesk.com/price/bitcoin"
            target="_blank"
          >
            Powered by CoinDesk
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Powered by CoinDesk",
            "href": "https://www.coindesk.com/price/bitcoin",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Powered by CoinDesk",
          "type": "a",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Wallet Balance
          </h2>,
          <div
            className="ui divider"
          />,
          <Connect(UserBalance) />,
          <div
            className="ui divider"
          />,
          <Connect(BitcoinBalance) />,
          <div>
            <a
              href="https://www.coindesk.com/price/bitcoin"
              target="_blank"
            >
              Powered by CoinDesk
            </a>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Wallet Balance",
          },
          "ref": null,
          "rendered": "Wallet Balance",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ui divider",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ui divider",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://www.coindesk.com/price/bitcoin"
              target="_blank"
            >
              Powered by CoinDesk
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Powered by CoinDesk",
              "href": "https://www.coindesk.com/price/bitcoin",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Powered by CoinDesk",
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
